beacon_node/eth2-libp2p/src/rpc/handler.rs

```diff
@@ -366,10 +374,27 @@ where
                 substream: out,
                 request,
             };
-            if let Some(_) = self
-                .outbound_substreams
-                .insert(id, (awaiting_stream, delay_key, protocol))
-            {
+            let response_chunk_count: Option<u64> = match protocol {
+                Protocol::BlocksByRange => {
+                    if let RPCRequest::BlocksByRange(req) = request {
+                        Some(req.count)
+                    } else {
+                        None
+                    }
+                }
+                Protocol::BlocksByRoot => {
+                    if let RPCRequest::BlocksByRoot(req) = request {
+                        Some(req.block_roots.len() as u64)
+                    } else {
+                        None
+                    }
+                }
+                _ => None,
+            };
```

Comment:
Here just match on the request. The protocol is taken from the request a couple of lines above so the `if let`s can't fail